microsoft365defender_connector.py

```diff
@@ -0,0 +1,1317 @@
+            self.debug_print("Exception Message - {}".format(str(e)))
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Error Connecting to server. Details: {0}"
+                                                   .format(self._get_error_message_from_exception(e))), resp_json)
+
+        return self._process_response(response, action_result)
+
+    def _get_asset_name(self, action_result):
+        """ Get name of the asset using Phantom URL.
+
+        :param action_result: object of ActionResult class
+        :return: status phantom.APP_ERROR/phantom.APP_SUCCESS(along with appropriate message), asset name
+        """
+
+        asset_id = self.get_asset_id()
+        rest_endpoint = DEFENDER_PHANTOM_ASSET_INFO_URL.format(asset_id=asset_id)
+        url = '{}{}'.format(DEFENDER_PHANTOM_BASE_URL.format(phantom_base_url=self.get_phantom_base_url()), rest_endpoint)
+        ret_val, resp_json = self._make_rest_call(action_result=action_result, endpoint=url, verify=False)
+
+        if phantom.is_fail(ret_val):
+            return ret_val, None
+
+        asset_name = resp_json.get('name')
+        if not asset_name:
+            return action_result.set_status(phantom.APP_ERROR, 'Asset Name for id: {0} not found.'.format(asset_id),
+                                            None)
+        return phantom.APP_SUCCESS, asset_name
+
+    def _get_phantom_base_url_defender(self, action_result):
+        """ Get base url of phantom.
+
+        :param action_result: object of ActionResult class
+        :return: status phantom.APP_ERROR/phantom.APP_SUCCESS(along with appropriate message),
+        base url of phantom
+        """
+
+        url = '{}{}'.format(DEFENDER_PHANTOM_BASE_URL.format(phantom_base_url=self.get_phantom_base_url()), DEFENDER_PHANTOM_SYS_INFO_URL)
+        ret_val, resp_json = self._make_rest_call(action_result=action_result, endpoint=url, verify=False)
+        if phantom.is_fail(ret_val):
+            return ret_val, None
+
+        phantom_base_url = resp_json.get('base_url').rstrip('/')
+        if not phantom_base_url:
+            return action_result.set_status(phantom.APP_ERROR, DEFENDER_BASE_URL_NOT_FOUND_MSG), None
+        return phantom.APP_SUCCESS, phantom_base_url
+
+    def _get_app_rest_url(self, action_result):
+        """ Get URL for making rest calls.
+
+        :param action_result: object of ActionResult class
+        :return: status phantom.APP_ERROR/phantom.APP_SUCCESS(along with appropriate message),
+        URL to make rest calls
+        """
+
+        ret_val, phantom_base_url = self._get_phantom_base_url_defender(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), None
+
+        ret_val, asset_name = self._get_asset_name(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), None
+
+        self.save_progress('Using Phantom base URL as: {0}'.format(phantom_base_url))
```

Comment:
All occurrences of `phantom` needs to be updated to `SOAR`, including variable names

Reply:
Hi @alexa-phantom,
We have followed the naming convention that is followed in the BaseConnector class. There also there is a method called "get_phantom_base_url()". Hence, we have kept the name of the variable as "phantom_base_url".

Reply:
Do you think we can change it to `Using SOAR base URL as` instead?

Reply:
Updated